Shared/Crypto.swift

```diff
@@ -120,60 +122,31 @@ final class Crypto {
     }
 
     func lookupKey(_ keyClass: KeyClass) throws -> SecKey {
-        var keyClassConstant = "" as CFString
-        switch keyClass {
-            case .private: keyClassConstant = kSecAttrKeyClassPrivate
-            case .public: keyClassConstant = kSecAttrKeyClassPublic
+        let keyClassConstant: CFString = switch keyClass {
+        case .private:
+            kSecAttrKeyClassPrivate
+        case .public:
+            kSecAttrKeyClassPublic
         }
 
         var query: [CFString: Any] = [
             kSecClass: kSecClassKey,
-            kSecMatchLimit: kSecMatchLimitAll,
+            kSecMatchLimit: kSecMatchLimitOne,
             kSecAttrAccessGroup: Info.groupId,
             kSecAttrKeyClass: keyClassConstant,
-            kSecReturnAttributes: true,
-            kSecReturnData: true
+            kSecReturnRef: true,
         ]
-
         query[kSecUseDataProtectionKeychain] = true
 
-        // Query the keychain
         var rawCopyResult: CFTypeRef? = nil
         let copyStatus = SecItemCopyMatching(query as CFDictionary, &rawCopyResult)
         guard copyStatus == errSecSuccess else {
             print("Human-readable error:", SecCopyErrorMessageString(copyStatus, nil) ?? "")
             throw CryptoError.fetchingKeys
         }
 
-        // Iterate through the results
-        let resultList = rawCopyResult! as! [[CFString: Any]]
-        for info in resultList {
-            guard let accessGroup = info[kSecAttrAccessGroup] else { continue }
-            guard accessGroup as! String == Info.groupId else { continue }
-            guard let resultClass = (info[kSecAttrKeyClass] as! CFString?) else { continue }
-            guard String(describing: resultClass) == String(describing: keyClassConstant) else { continue }
-            guard let keyData = (info[kSecValueData] as! CFData?) else { continue }
-
-            // We have a match. Turn it into a SecKey we can use.
-            // (Irritatingly, "synchronizable" keys don't give you SecKey objects...)
-            let matchedKeyAttributes: [CFString: Any] = [
-                kSecClass: kSecClassKey,
-                kSecAttrKeyClass: keyClassConstant,
-                kSecAttrKeyType: constants["type"]!,
-                kSecAttrKeySizeInBits: constants["bits"]!
-            ]
-
-            var secKeyError: Unmanaged<CFError>? = nil
-            guard let key = SecKeyCreateWithData(keyData, matchedKeyAttributes as CFDictionary, &secKeyError) else {
-                print(secKeyError!.takeUnretainedValue() as Error)
-                throw CryptoError.fetchingKeys
-            }
-
-            return key
-        }
-
-        // If we get to this point, there are no matches.
-        throw CryptoError.fetchingKeys
```

Comment:
Another reason I believe my changes are effective is this section. Initially, I refactored it, but then I removed the result passing completely.

Additionally, I noticed that when I encountered the CryptoError, the Whitelist was functioning perfectly, and Shut Up was able to decrypt it, despite the error message. Therefore, I suspect that during the parsing process, something went awry, and Crypto.Error was thrown, even though it shouldn’t have been.